core/arch/arm/kernel/secure_partition.c

```diff
@@ -426,16 +415,11 @@ static TEE_Result check_fdt(const void * const fdt, const TEE_UUID *uuid)
 	if (description)
 		DMSG("Loading SP: %s", description->data);
 
-	if (sp_dt_get_uuid(fdt, 0, "uuid", &fdt_uuid)) {
+	if (!uuid ||  sp_dt_get_uuid(fdt, 0, "uuid", uuid)) {
```

Comment:
Is it meaningful to supply `uuid` as NULL? I guess not since `sp_init_uuid()` always gives a valid pointer.

Reply:
Ack